lib/rules/lint-eol-last.js

```diff
@@ -0,0 +1,59 @@
+'use strict';
+
+const Rule = require('./base');
+
+module.exports = class EolLast extends Rule {
+  visitor() {
+    return {
+      Program: {
+        exit(node) {
+          let line = node.loc.start.line;
+          let column = node.loc.start.column;
+
+          // yielded content makes it in here too
+          //
+          // `{{#my-component}}
+          //   test
+          // {{/my-component}}`
+          //
+          // becomes
+          //
+          // `
+          //   test
+          // `
+          //
+          // check for that case
+          if (line !== 1 || column !== 0) {
+            return;
+          }
+
+          let source = this.sourceForNode(node);
+
+          let lastChar = source[source.length - 1];
+          let message;
+          switch (this.config) {
+          case 'always':
+            if (lastChar !== '\n') {
```

Comment:
Shouldn't we allow the EOL char(s) to be up to the user?  Some folks want `\r\n` (two chars, I know), and others want `\n`.  We _could_ leave this up to the `linebreak-style` rule maybe?

I ultimately want the default line ending to be auto-detected from `.editorconfig` (see https://github.com/rwjblue/ember-template-lint/issues/284), but that doesn't need to block this rule...

Reply:
This is already agnostic, whether the EOL is '\r\n' or '\n', the last char of the string will always be '\n'. (except old mac where it is only '\r'.......)